app/src/keymap.c

```diff
@@ -118,24 +122,46 @@ uint8_t zmk_keymap_highest_layer_active() {
     return zmk_keymap_layer_default();
 }
 
-int zmk_keymap_layer_activate(uint8_t layer) { return set_layer_state(layer, true); };
+int zmk_keymap_layer_activate(uint8_t layer) {
+    zmk_keymap_layers_state_t prior_state = _zmk_keymap_layer_state;
+
+    int ret = set_layer_state(layer, true);
+    raise_layer_event(prior_state);
+
+    return ret;
+}
+
+int zmk_keymap_layer_deactivate(uint8_t layer) {
+    zmk_keymap_layers_state_t prior_state = _zmk_keymap_layer_state;
 
-int zmk_keymap_layer_deactivate(uint8_t layer) { return set_layer_state(layer, false); };
```

Comment:
I think `zmk_keymap_layer_deactivate` and `zmk_keymap_layer_activate` are good to keep around.

Reply:
They are still there.